File: evidence/generated-systems/angular/aerospace-on-angular/src/app/models/Plant.js

// Define collection and schema for Plant
export interface Plant {
    name:
	type : string
    plantCode:
	type : string
    address:
	type : Address
    Manufacturer:
	type : Schema.Types.ObjectId
    ProductionLines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ProductionLine' }]
    Warehouses:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Warehouse' }]
#
    collection: 'plants'
}

File: evidence/generated-systems/rubyonrails/aerospace-on-rails/aerospaceonrails/app/models/plant.rb
class Plant < ApplicationRecord


  composed_of :address,
    class_name: "Address",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      %w[address_country country]
    ]

  has_many :Manufacturer, class_name: 'AerospaceManufacturer'
  has_many :ProductionLines, class_name: 'ProductionLine'
  has_many :Warehouses, class_name: 'Warehouse'

end
